bundles/org.openhab.binding.panamaxfurman/src/main/java/org/openhab/binding/panamaxfurman/internal/PanamaxFurmanAbstractHandler.java

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.panamaxfurman.internal;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.panamaxfurman.internal.protocol.PowerConditionerChannel;
+import org.openhab.binding.panamaxfurman.internal.protocol.ProtocolMapper;
+import org.openhab.binding.panamaxfurman.internal.protocol.event.PanamaxFurmanCommunicationEventListener;
+import org.openhab.binding.panamaxfurman.internal.protocol.event.PanamaxFurmanConnectionActiveEvent;
+import org.openhab.binding.panamaxfurman.internal.protocol.event.PanamaxFurmanConnectionBrokenEvent;
+import org.openhab.binding.panamaxfurman.internal.protocol.event.PanamaxFurmanConnectivityEvent;
+import org.openhab.binding.panamaxfurman.internal.protocol.event.PanamaxFurmanInformationReceivedEvent;
+import org.openhab.binding.panamaxfurman.internal.transport.PanmaxFurmanConnector;
+import org.openhab.binding.panamaxfurman.internal.util.TimeInterval;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PanamaxFurmanHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Dave Badia - Initial contribution
+ */
+@NonNullByDefault
+public abstract class PanamaxFurmanAbstractHandler extends BaseThingHandler
+        implements PanamaxFurmanCommunicationEventListener {
+    private final Logger logger = LoggerFactory.getLogger(PanamaxFurmanAbstractHandler.class);
+
+    private static final long AVOID_DUPLICATE_REQUESTS_WITHIN_MILLIS = TimeUnit.SECONDS.toMillis(3);
+    private static final Map<String, Long> REQUEST_LAST_SENT_AT = new ConcurrentHashMap<>();
+    private final ProtocolMapper protocolMapper;
+    private @NonNullByDefault({}) PanmaxFurmanConnector connector;
```

Comment:
`@NonNullByDefault({})` can be replaced by `@Nullable` as you are allready checking this property for null at several places

Reply:
Eclipse shows a warning when I change it to @Nullable.  Let me commit and see what the build server thinks